pkg/wasm/monitoring.go

```diff
@@ -0,0 +1,93 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package wasm
+
+import "istio.io/pkg/monitoring"
+
+type fetchStatus int
+
+const (
+	fetchSuccess fetchStatus = iota
+	downloadFailure
+	checksumMismatch
+)
+
+type conversionStatus int
+
+const (
+	conversionSuccess conversionStatus = iota
+	noRemoteLoad
+	marshalFailure
+	fetchFailure
+	missRemoteFetchHint
+)
+
+var (
+	hitTag    = monitoring.MustCreateLabel("hit")
+	resultTag = monitoring.MustCreateLabel("result")
+
+	fetchResultLabelMap = map[fetchStatus]string{
```

Comment:
Curious: why did you decide to not simplify define tags as strings, and use enums? Not that it matters, you could probably override define String() on custom int type to avoid map lookup.

Reply:
No particular reason. I simplified it to const string directly. Thanks!